source/slang/emit.cpp

```diff
@@ -5612,9 +5612,20 @@ struct EmitVisitor
                             {
                             default: Emit("rgba");  break;
 
-                            // TODO: GLSL doesn't actually seem to support 3-component formats
-                            // universally, so this might cause problems.
-                            case 3:  Emit("rgb");   break;
+                            case 3:
+                            {
+                                // TODO: GLSL doesn't actually seem to support 3-component formats
+                                // universally so for now we are going to default to rgba
+                                //
+                                // It seems that SPIR-V can support such formats, so with direct SPIR-V
+                                // path, and/or with support glslang could support in the future.
```

Comment:
I think this comment is incorrect. I just looked at [the SPIR-V spec](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.pdf) section 3.11 on Image Formats and it doesn't list an `rgb32f`.

What it seems like SPIR-V *does* support is having an image with an unknown-at-compile-time format, so long as the underlying API supports it (and I hope Vulkan is the API they mean). Ideally that would mean that we can just drop all these qualifiers when emitting GLSL for Vulkan targets.